### ISSUE TITLE
ExpressCheckout IPN response is "INVALID"

### ISSUE THREAD

**Issue by reymark-torres** [srmklive/laravel-paypal#233]
I've been using this plugin for quite some time now. But according to PayPal IPN integration guide, we should now send the response messages back to PayPal on the following URLs:

- https://ipnpb.sandbox.paypal.com/cgi-bin/webscr (for Sandbox IPNs)
- https://ipnpb.paypal.com/cgi-bin/webscr (for live IPNs)

I tried the old URL which is https://sandbox.paypal.com/cgi-bin/webscr and the new URL https://ipnpb.sandbox.paypal.com/cgi-bin/webscr but I still get an "INVALID" response with "Completed" payment_status

Where did I go wrong?

Your help is much appreciated.

Source: https://developer.paypal.com/docs/classic/ipn/integration-guide/IPNImplementation/

**Comment by srmklive** [comment]
Hi,

My apologies for responding really late on this issue. I got to troubleshoot this issue on the demo site https://laravel-paypal-demo.srmk.info/. It took a bit of time to figure out the possible issue with IPN. So here it is:

![screenshot from 2018-08-21 23-34-41](https://user-images.githubusercontent.com/839335/44423095-918aab00-a59e-11e8-983a-9db2e417b1a3.png)

Look at the last two records of the tinker output. It shows two different IPN status for same invoice. Turns out initial request to PayPal for SetExpressCheckout which returns the token is also returned by PayPal through IPN initially. When transaction is completed, another IPN verification call returns VERIFIED status.
